conans/client/command.py

```diff
@@ -1821,67 +1829,83 @@ def frogarian(self, *args):
         """
         cmd_frogarian(self._out)
 
-    def graph(self, *args):
+    def lock(self, *args):
         """
         Generates and manipulates lock files.
         """
-        parser = argparse.ArgumentParser(description=self.graph.__doc__,
-                                         prog="conan graph",
+        parser = argparse.ArgumentParser(description=self.lock.__doc__,
+                                         prog="conan lock",
                                          formatter_class=SmartFormatter)
         subparsers = parser.add_subparsers(dest='subcommand', help='sub-command help')
         subparsers.required = True
 
         # create the parser for the "a" command
-        merge_cmd = subparsers.add_parser('update-lock', help='merge two lockfiles')
-        merge_cmd.add_argument('old_lockfile', help='path to previous lockfile')
-        merge_cmd.add_argument('new_lockfile', help='path to modified lockfile')
+        update_cmd = subparsers.add_parser('update', help='update a lockfile with another lockfile')
+        update_cmd.add_argument('old_lockfile', help='path to previous lockfile')
+        update_cmd.add_argument('new_lockfile', help='path to modified lockfile')
```

Comment:
Maybe we can consider the option: 

`conan update app.lock --lockfile=liba.lock --lockfile-out=another_name.lock `
Defaulting `--lockfile-out` as `app.lock` if not passed.

Also for the help I would add a little bit more information, maybe something like this:

'path to previous lockfile' -> 'Path to the lockfile that is going to be updated.'
'path to modified lockfile' -> 'Path to the lockfile with the information that is going to be incorporated in the input lockfile.'

Reply:
The truth is that we have been violating good cli design, at least as recommended by argparse:

https://docs.python.org/3/library/argparse.html#required

> Note Required options are generally considered bad form because users expect options to be optional, and thus they should be avoided when possible.

Reply:
Thinking it twice, it makes sense to expect options to be optional, then let's remove the `--lockfile-out` and default to `conan.lock` as well. And just adding a bit more of info to the help message would be enough for the `update` sub-command.